pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2227,3 +2240,12 @@ def test_categorical_non_unique_monotonic(n_categories):
         index=left_index,
     )
     tm.assert_frame_equal(expected, result)
+
+
+def test_right_index_true_right_join_target_index():
+    df_left = pd.DataFrame(index=["a", "b"])
+    df_right = pd.DataFrame({"x": ["a", "c"]})
+
+    result = pd.merge(df_left, df_right, left_index=True, right_on="x", how="left")
+    expected = pd.DataFrame({"x": ["a", "b"]}, index=Index(["a", "b"]))
+    tm.assert_frame_equal(result, expected)
```

Comment:
Do you mean "left" instead of "right" in the test name? I'd also include a link to the issue as a comment, and rename df_left -> left, df_right -> right to be consistent with other tests.

How does this behave when we actually are doing a right join? It feels like it should exactly equal the right DataFrame (index and values), but someone who knows more than me would have to chime in.

Reply:
Yeah you are right of course. It was late yesterday evening :)

I added a test to show the behavior in case of a right join. I am not quite sure what to expect. The current behavior is the result you described. But the documentation says, that indexes are passed along if used in join. I am not sure how to interpret this in case of right join when left index is used.